### ISSUE TITLE
Anyone using Sonoff?

### ISSUE THREAD

**Issue by wilberforce** [Moddable-OpenSource/moddable#176]
I've just got my hands on a 1 ch sonoff and got it working usng the factory firmware.
Next challenge is to get moddable onto it .
I think it uses the esp8266 variant and 1mb of flash. 
Before I solder headers and and FTDI-> usb thingy, (which It think will be a bit of phff to flash as it will need the glash button to be hit at power up) I'm wondering if anyone has got ota esp8266 update moddable working - I remember seeing esp32 code - not sure about the esp826?

 With only 1mb I guess the map would have to be a 512/512 for ota , size size might be a bit of a squeeze and I might be dreaming. Sorry this is a bit of a rant  - however there isn't a forum to discuss such things!

Oh - and can C++ libraries be used with moddable? I'm looking to port a Dallas Onewire to the esp8266 - however the auduino one is C++.

**Comment by jim80** [comment]
" I'm looking to port a Dallas Onewire to the esp8266 "
Check the current pull request for this.

**Comment by wilberforce** [comment]
@jim80

You mean the pull request I opened :-;

https://github.com/Moddable-OpenSource/moddable/pull/170

That's for the esp32 - I'm working on the esp8266 port now

**Comment by jim80** [comment]
Ha! Ooops. Please forgive my lack of attention!
I'll be pleased to see the 8266 port, many thanks for that.

**Comment by wilberforce** [comment]
Using this manifest fragment, the sonoff is working!

```
	"esp/8285": {
	        "build": {
                "FLASH_SIZE": "1M",
                "FLASH_LAYOUT": "eagle.flash.1m.ld"
				},
			"config": {
				"led1_pin": 13,
				"button1_pin": 0,
			},
			"modules": {
				"*": "$(MODULES)/pins/digital/monitor/esp/*",
			},
		},	
```

![image](https://user-images.githubusercontent.com/5050824/57190925-25a57280-6f74-11e9-89fc-e6b98fa4de24.png)

**Comment by phoddie** [comment]
Cool. Congrats. Apologies for not replying sooner. That linker script is in the project from work we did on Sonoff products last year. We haven't documented it yet, but it does work. ;)

I gave a couple talks about hacking the Sonoff bulb earlier this year, at an IoT event at MIT and a Scripts and Things meetup. I'll be [presenting it at Maker Faire](https://makerfaire.com/maker/entry/69518/) in a couple weeks.
